Oleksii task

Sir-Beard opened the pull request:
Please review my project.

Sir-Beard committed:
test

Sir-Beard committed:
Test commit 07122022 1955

Rommelua commented on src/main/java/core/mate/academy/model/Bulldozer.java:
```diff
@@ -5,9 +5,43 @@
  * Do not remove no-args constructor
  */
 public class Bulldozer extends Machine {
+    private String typeOfBulldozer; //Heavy bulldozer
+    private String nameOfBu; //Caterpillar
+    private int weightOfBu; //48,988kg
```
Не нужно оставлять бесполезные комментарии. Мы относимся к коду так, как будто коммитим его в рабочий проект.

Rommelua commented on src/main/java/core/mate/academy/model/Excavator.java:
```diff
@@ -5,7 +5,42 @@
+    private int weightOfExcavator; //48,988kg
+
     public Excavator() {
+
```
empty line

Rommelua commented on src/main/java/core/mate/academy/model/Truck.java:
```diff
@@ -5,11 +5,47 @@
+    private int weightOfTruck; //48,988kg
+
     public Truck() {
+
```
empty line

Rommelua commented on src/main/java/core/mate/academy/service/BulldozerProducer.java:
```diff
@@ -0,0 +1,17 @@
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Bulldozer> {
+    //@Override
```
```suggestion
    @Override
```

Rommelua commented on src/main/java/core/mate/academy/service/MachineServiceImpl.java:
```diff
@@ -1,7 +1,46 @@
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
```
Лучше сделать продьюсеры переменными класса, а не создавать их каждый раз при обращении к методу

Rommelua commented on src/main/java/core/mate/academy/service/MachineServiceImpl.java:
```diff
@@ -1,7 +1,46 @@
+        if (type.equals(Excavator.class)) {
+            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+            List<Excavator> excavatorList = excavatorProducer.get();
+
```
empty line

Rommelua commented on src/main/java/core/mate/academy/service/MachineServiceImpl.java:
```diff
@@ -1,7 +1,46 @@
+        if (type.equals(Bulldozer.class)) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            List<Bulldozer> bulldozerList = bulldozerProducer.get();
+
```
empty line

Rommelua commented on src/main/java/core/mate/academy/service/MachineServiceImpl.java:
```diff
@@ -1,7 +1,46 @@
+            return new ArrayList<>(truckList);
+        }
+
+        return /*Collections.emptyList()*/ new ArrayList<>();
```
comment

Rommelua commented on src/main/java/core/mate/academy/service/MachineServiceImpl.java:
```diff
@@ -1,7 +1,46 @@
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> machines) {
+    }
```
where is the body?

Sir-Beard committed:
Test commit 08122022 1850

Sir-Beard committed:
Test commit 08122022 1855

Sir-Beard committed:
Test commit 08122022 1918

Rommelua commented on src/main/java/core/mate/academy/service/MachineServiceImpl.java:
```diff
@@ -1,7 +1,47 @@
+
+    @Override
+    public void startWorking(List<? extends Machine> machines) {
+        Machine machine = machines.get(0);
+        machine.doWork();
```
???
![image](https://user-images.githubusercontent.com/20326858/206718844-b6963610-2cb3-48ad-86d6-84f7eed96a7f.png)

Sir-Beard committed:
Commit 10122022 1726
Method "startWorking" rewritten.

Sir-Beard committed:
Commit 11122022 1842
Method "startWorking" now have for loop, where each machine invoke "doWork" method for every exactly machine type.
